autogluon/task/tabular_prediction/predictor.py

```diff
@@ -341,14 +353,12 @@ def _summarize(key, msg, results):
 
     @staticmethod
     def __get_dataset(dataset):
-        if isinstance(dataset, TabularDataset):
-            return dataset
         if isinstance(dataset, str):
-            return TabularDataset(file_path=dataset)
+            return pd.read_csv(dataset)
```

Comment:
Use load_pd.load(). pd.read_csv introduces many subtle issues and does not support other file types such as parquet. If you are going to use it (which we won't), you need to specify low_memory=False, otherwise dtypes are inferred in batches of rows, causing massive problems. In general, never use pd.read_csv() to load data in AutoGluon.

Reply:
ok

Reply:
Where is the function load_pd? could you post a link for more info about the issue you describe?

Reply:
http://acepor.github.io/2017/09/19/pandas-dtype/

Reply:
autogluon/utils/tabular/utils/loaders/load_pd.py